charm/kafka/reactive/kafka.py

```diff
@@ -355,6 +360,23 @@ def import_ca_crt_to_keystore():
             remove_state('tls_client.ca_installed')
             set_state('kafka.ca.keystore.saved')
 
+@hook('upgrade-status')
+@when('kafka.started')
+def health_check():
+    # Health check will attempt to restart Kafka service, if the service is not running
+    kafka = Kafka()
+    if kafka.is_running():
+        return
+
+    for i in range(3):
+        hookenv.status_set('maintenance', 'attempting to restart kafka, attempt: {}'.format(i))
+        kafka.restart()
+        if kafka.is_running():
+            return
```

Comment:
Need to set status to active here, to clear prior statuses like mainteance, blocked.